engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/TaskEntity.java

```diff
@@ -1451,6 +1453,10 @@ public void initializeFormKey() {
       }
     }
   }
+  public void initializeAttachmentAndComments(){
+    this.attachmentExists = !getProcessEngine().getTaskService().getTaskAttachments(id).isEmpty();
+    this.commentExists = !getProcessEngine().getTaskService().getTaskComments(id).isEmpty();
+  }
```

Comment:
Note to myself:
* Shouldn't we try to fetch data from cache similar to `#initializeFormKey()` (`#getTaskDefinition()`)?
* Evaluate again possible options for this data. Do we need to have the changes in Java API or we can have the data fetch in REST API? For consistency, we can have them in both APIs.